src/components /Calendar.jsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState, useEffect } from "react";
+
+import { Calendar, momentLocalizer } from "react-big-calendar";
+import moment from "moment";
+import "react-big-calendar/lib/css/react-big-calendar.css";
+
+import { Fragment, useRef } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { ExclamationTriangleIcon } from "@heroicons/react/24/outline";
+
+import uuid from "react-uuid";
+
+const CalendarComponent = (props) => {
+  // console.log(props);
+  const data = props.medicineData;
+  const moodData = props.moodData;
+  // console.log (data);
+  const localizer = momentLocalizer(moment);
+  const [events, setEvents] = useState([]);
+
+  const [selectedEvent, setSelectedEvent] = useState(undefined);
+  const [displayModal, setDisplayModal] = useState(false);
+
+  const [show, setShow] = useState(false);
+
+  let tempEvents = [];
+
+  useEffect(() => {
```

Comment:
This whole effect seems a bit inefficient to me, and it might have to do with the architecture of separating the medicine and mood events. These events are decoupled, but somehow still coupled in the end. I think this might be worth rethinking the whole data structure for these events. Since both of them actually carry the same values, we might be able to unify these two?